src/foremast/awslambda/awslambda.py

```diff
@@ -57,7 +57,9 @@ def __init__(self, app, env, region, prop_path):
         self.memory = self.properties[env]['app']['lambda_memory']
         self.timeout = self.properties[env]['app']['lambda_timeout']
 
-        self.role_arn = get_role_arn(generated.iam()['role'], self.env, self.region)
+        self.role_arn = get_role_arn(
+            self.properties[env]['app'].get('instance_profile', generated.iam()['role']),
```

Comment:
`profile` and `role` are two different words. I predict there will be an issue here. We must double check.